src/zenml/new/steps/step_decorator.py

```diff
@@ -0,0 +1,181 @@
+#  Copyright (c) ZenML GmbH 2021. All Rights Reserved.
```

Comment:
Answer: the difference is a single bracket, it returns an instance instead of a step class.

Reply:
This means there's a lot of code duplication, but we decided that's fine for now since the old decorator is deprecated and will be deleted soon enough anyways